src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/service.utils.ts

```diff
@@ -101,28 +101,31 @@ export const getManifestData = ({
         configType === AppEnvDeploymentConfigType.PUBLISHED_WITH_DRAFT
     const isDefaultSelected = configType === AppEnvDeploymentConfigType.DEFAULT_VERSION
 
-    const deploymentManifestRequestData: Record<string, string | number> = {
+    const isHistoricData = !values && envId > -1
+
+    const deploymentManifestRequestData: GetDeploymentManifestProps = {
         appId: +appId,
-        valuesAndManifestFlag: 2,
         chartRefId: manifestChartRefId,
+        ...(envId > -1 && {
+            envId,
+        }),
+        ...(!isDefaultSelected && {
+            ...(isHistoricData
+                ? {
+                      type:
+                          identifierId && pipelineId
+                              ? TemplateListType.DeployedOnSelfEnvironment
+                              : TemplateListType.PublishedOnEnvironments,
+                      deploymentTemplateHistoryId: identifierId,
+                      pipelineId,
+                  }
+                : {
+                      values: values || null,
+                  }),
+        }),
     }
 
-    if (envId > -1) {
-        deploymentManifestRequestData.envId = envId
-
-        if (!values && !isDefaultSelected) {
-            deploymentManifestRequestData.type =
-                identifierId && pipelineId
-                    ? TemplateListType.DeployedOnSelfEnvironment
-                    : TemplateListType.PublishedOnEnvironments
-            deploymentManifestRequestData.deploymentTemplateHistoryId = identifierId
-            deploymentManifestRequestData.pipelineId = pipelineId
-        }
-    }
-
-    if (values && !isDefaultSelected) {
-        deploymentManifestRequestData.values = values
-    }
-
-    return !isDraftSelected || values ? getDeploymentManisfest(deploymentManifestRequestData) : nullResponse
+    return !isDraftSelected || values
+        ? getDeploymentManifest(deploymentManifestRequestData)
+        : Promise.resolve(nullResponse)
```

Comment:
Yes otherwise method will assume type:``` ResponseType<ResolvedDeploymentTemplateDTO> | Promise<ResponseType<ResolvedDeploymentTemplateDTO>>``` instead of ```ResponseType<ResolvedDeploymentTemplateDTO>```